### PR TITLE
Add endpoint listing AWS sizes

### PR DESCRIPTION
**Pull request by maciaszczykm**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*: Related to https://github.com/kubermatic/dashboard-v2/issues/1245.

**Special notes for your reviewer**: Example output from `127.0.0.1:8080/api/v1/providers/aws/sizes`:

```json
[
    {
        "name": "c5d.xlarge",
        "pretty_name": "C5 High-CPU Extra Large",
        "memory": 8,
        "vcpus": 4
    },
    {
        "name": "m5a.2xlarge",
        "pretty_name": "M5A Double Extra Large",
        "memory": 32,
        "vcpus": 8
    },
    {
        "name": "c5.9xlarge",
        "pretty_name": "C5 High-CPU 9xlarge",
        "memory": 72,
        "vcpus": 36
    },
    {
        "name": "r5ad.xlarge",
        "pretty_name": "R5AD Extra Large",
        "memory": 32,
        "vcpus": 4
    },
    {
        "name": "c5.large",
        "pretty_name": "C5 High-CPU Large",
        "memory": 4,
        "vcpus": 2
    },
    {
        "name": "c5n.large",
        "pretty_name": "C5N Large",
        "memory": 5.25,
        "vcpus": 2
    },
    {
        "name": "i2.xlarge",
        "pretty_name": "I2 Extra Large",
        "memory": 30.5,
        "vcpus": 4
    },
    {
        "name": "i3en.3xlarge",
        "pretty_name": "I3EN 3xlarge",
        "memory": 96,
        "vcpus": 12
    },
    {
        "name": "z1d.3xlarge",
        "pretty_name": "Z1D 3xlarge",
        "memory": 96,
        "vcpus": 12
    },
    {
        "name": "m5.2xlarge",
        "pretty_name": "M5 General Purpose Double Extra Large",
        "memory": 32,
        "vcpus": 8
    },
    {
        "name": "i2.2xlarge",
        "pretty_name": "I2 Double Extra Large",
        "memory": 61,
        "vcpus": 8
    },
    {
        "name": "i3en.2xlarge",
        "pretty_name": "I3EN Double Extra Large",
        "memory": 64,
        "vcpus": 8
    },
    {
        "name": "m5a.xlarge",
        "pretty_name": "M5A Extra Large",
        "memory": 16,
        "vcpus": 4
    },
    {
        "name": "t2.2xlarge",
        "pretty_name": "T2 Double Extra Large",
        "memory": 32,
        "vcpus": 8
    },
    {
        "name": "t2.large",
        "pretty_name": "T2 Large",
        "memory": 8,
        "vcpus": 2
    },
    {
        "name": "r5a.large",
        "pretty_name": "R5A Large",
        "memory": 16,
        "vcpus": 2
    },
    {
        "name": "c3.large",
        "pretty_name": "C3 High-CPU Large",
        "memory": 3.75,
        "vcpus": 2
    },
    {
        "name": "a1.2xlarge",
        "pretty_name": "A1 Double Extra Large",
        "memory": 16,
        "vcpus": 8
    },
    {
        "name": "c4.xlarge",
        "pretty_name": "C4 High-CPU Extra Large",
        "memory": 7.5,
        "vcpus": 4
    },
    {
        "name": "m5ad.xlarge",
        "pretty_name": "M5AD Extra Large",
        "memory": 16,
        "vcpus": 4
    },
    {
        "name": "m4.large",
        "pretty_name": "M4 General Purpose Large",
        "memory": 8,
        "vcpus": 2
    },
    {
        "name": "h1.4xlarge",
        "pretty_name": "H1 Quadruple Extra Large",
        "memory": 64,
        "vcpus": 16
    },
    {
        "name": "x1e.xlarge",
        "pretty_name": "X1E Extra Large",
        "memory": 122,
        "vcpus": 4
    },
    {
        "name": "m5.large",
        "pretty_name": "M5 General Purpose Large",
        "memory": 8,
        "vcpus": 2
    },
    {
        "name": "c5.4xlarge",
        "pretty_name": "C5 High-CPU Quadruple Extra Large",
        "memory": 32,
        "vcpus": 16
    },
    {
        "name": "r3.large",
        "pretty_name": "R3 High-Memory Large",
        "memory": 15.25,
        "vcpus": 2
    },
    {
        "name": "c4.large",
        "pretty_name": "C4 High-CPU Large",
        "memory": 3.75,
        "vcpus": 2
    },
    {
        "name": "r5d.xlarge",
        "pretty_name": "R5D Extra Large",
        "memory": 32,
        "vcpus": 4
    },
    {
        "name": "m5d.large",
        "pretty_name": "M5 General Purpose Large",
        "memory": 8,
        "vcpus": 2
    },
    {
        "name": "r5.2xlarge",
        "pretty_name": "R5 Double Extra Large",
        "memory": 64,
        "vcpus": 8
    },
    {
        "name": "m3.2xlarge",
        "pretty_name": "M3 General Purpose Double Extra Large",
        "memory": 30,
        "vcpus": 8
    },
    {
        "name": "m5d.2xlarge",
        "pretty_name": "M5 General Purpose Double Extra Large",
        "memory": 32,
        "vcpus": 8
    },
    {
        "name": "m5ad.large",
        "pretty_name": "M5AD Large",
        "memory": 8,
        "vcpus": 2
    },
    {
        "name": "c5d.large",
        "pretty_name": "C5 High-CPU Large",
        "memory": 4,
        "vcpus": 2
    },
    {
        "name": "c5n.2xlarge",
        "pretty_name": "C5N Double Extra Large",
        "memory": 21,
        "vcpus": 8
    },
    {
        "name": "m5d.xlarge",
        "pretty_name": "M5 General Purpose Extra Large",
        "memory": 16,
        "vcpus": 4
    },
    {
        "name": "m3.large",
        "pretty_name": "M3 General Purpose Large",
        "memory": 7.5,
        "vcpus": 2
    },
    {
        "name": "c5d.4xlarge",
        "pretty_name": "C5 High-CPU Quadruple Extra Large",
        "memory": 32,
        "vcpus": 16
    },
    {
        "name": "r4.large",
        "pretty_name": "R4 High-Memory Large",
        "memory": 15.25,
        "vcpus": 2
    },
    {
        "name": "r3.xlarge",
        "pretty_name": "R3 High-Memory Extra Large",
        "memory": 30.5,
        "vcpus": 4
    },
    {
        "name": "c5n.xlarge",
        "pretty_name": "C5N Extra Large",
        "memory": 10.5,
        "vcpus": 4
    },
    {
        "name": "m5a.4xlarge",
        "pretty_name": "M5A Quadruple Extra Large",
        "memory": 64,
        "vcpus": 16
    },
    {
        "name": "r5ad.2xlarge",
        "pretty_name": "R5AD Double Extra Large",
        "memory": 64,
        "vcpus": 8
    },
    {
        "name": "t2.xlarge",
        "pretty_name": "T2 Extra Large",
        "memory": 16,
        "vcpus": 4
    },
    {
        "name": "c3.8xlarge",
        "pretty_name": "C3 High-CPU Eight Extra Large",
        "memory": 60,
        "vcpus": 32
    },
    {
        "name": "m3.medium",
        "pretty_name": "M3 General Purpose Medium",
        "memory": 3.75,
        "vcpus": 1
    },
    {
        "name": "x1e.2xlarge",
        "pretty_name": "X1E Double Extra Large",
        "memory": 244,
        "vcpus": 8
    },
    {
        "name": "m5a.8xlarge",
        "pretty_name": "M5A Eight Extra Large",
        "memory": 128,
        "vcpus": 32
    },
    {
        "name": "m5ad.2xlarge",
        "pretty_name": "M5AD Double Extra Large",
        "memory": 32,
        "vcpus": 8
    },
    {
        "name": "r5a.2xlarge",
        "pretty_name": "R5A Double Extra Large",
        "memory": 64,
        "vcpus": 8
    },
    {
        "name": "d2.xlarge",
        "pretty_name": "D2 Extra Large",
        "memory": 30.5,
        "vcpus": 4
    },
    {
        "name": "c5.2xlarge",
        "pretty_name": "C5 High-CPU Double Extra Large",
        "memory": 16,
        "vcpus": 8
    },
    {
        "name": "t3.medium",
        "pretty_name": "T3 Medium",
        "memory": 4,
        "vcpus": 2
    },
    {
        "name": "m1.xlarge",
        "pretty_name": "M1 General Purpose Extra Large",
        "memory": 15,
        "vcpus": 4
    },
    {
        "name": "c5d.2xlarge",
        "pretty_name": "C5 High-CPU Double Extra Large",
        "memory": 16,
        "vcpus": 8
    },
    {
        "name": "m3.xlarge",
        "pretty_name": "M3 General Purpose Extra Large",
        "memory": 15,
        "vcpus": 4
    },
    {
        "name": "m1.medium",
        "pretty_name": "M1 General Purpose Medium",
        "memory": 3.75,
        "vcpus": 1
    },
    {
        "name": "t3a.medium",
        "pretty_name": "T3A Medium",
        "memory": 4,
        "vcpus": 2
    },
    {
        "name": "i3.xlarge",
        "pretty_name": "I3 High I/O Extra Large",
        "memory": 30.5,
        "vcpus": 4
    },
    {
        "name": "z1d.2xlarge",
        "pretty_name": "Z1D Double Extra Large",
        "memory": 64,
        "vcpus": 8
    },
    {
        "name": "c3.xlarge",
        "pretty_name": "C3 High-CPU Extra Large",
        "memory": 7.5,
        "vcpus": 4
    },
    {
        "name": "c3.2xlarge",
        "pretty_name": "C3 High-CPU Double Extra Large",
        "memory": 15,
        "vcpus": 8
    },
    {
        "name": "r3.2xlarge",
        "pretty_name": "R3 High-Memory Double Extra Large",
        "memory": 61,
        "vcpus": 8
    },
    {
        "name": "r4.2xlarge",
        "pretty_name": "R4 High-Memory Double Extra Large",
        "memory": 61,
        "vcpus": 8
    },
    {
        "name": "m5.xlarge",
        "pretty_name": "M5 General Purpose Extra Large",
        "memory": 16,
        "vcpus": 4
    },
    {
        "name": "c4.8xlarge",
        "pretty_name": "C4 High-CPU Eight Extra Large",
        "memory": 60,
        "vcpus": 36
    },
    {
        "name": "d2.2xlarge",
        "pretty_name": "D2 Double Extra Large",
        "memory": 61,
        "vcpus": 8
    },
    {
        "name": "m5ad.4xlarge",
        "pretty_name": "M5AD Quadruple Extra Large",
        "memory": 64,
        "vcpus": 16
    },
    {
        "name": "c5.xlarge",
        "pretty_name": "C5 High-CPU Extra Large",
        "memory": 8,
        "vcpus": 4
    },
    {
        "name": "m5d.4xlarge",
        "pretty_name": "M5 General Purpose Quadruple Extra Large",
        "memory": 64,
        "vcpus": 16
    },
    {
        "name": "m4.xlarge",
        "pretty_name": "M4 General Purpose Extra Large",
        "memory": 16,
        "vcpus": 4
    },
    {
        "name": "r5a.4xlarge",
        "pretty_name": "R5A Quadruple Extra Large",
        "memory": 128,
        "vcpus": 16
    },
    {
        "name": "t3.xlarge",
        "pretty_name": "T3 Extra Large",
        "memory": 16,
        "vcpus": 4
    },
    {
        "name": "m4.4xlarge",
        "pretty_name": "M4 General Purpose Quadruple Extra Large",
        "memory": 64,
        "vcpus": 16
    },
    {
        "name": "r4.xlarge",
        "pretty_name": "R4 High-Memory Extra Large",
        "memory": 30.5,
        "vcpus": 4
    },
    {
        "name": "a1.xlarge",
        "pretty_name": "A1 Extra Large",
        "memory": 8,
        "vcpus": 4
    },
    {
        "name": "r5d.4xlarge",
        "pretty_name": "R5D Quadruple Extra Large",
        "memory": 128,
        "vcpus": 16
    },
    {
        "name": "p2.xlarge",
        "pretty_name": "General Purpose GPU Extra Large",
        "memory": 61,
        "vcpus": 4
    },
    {
        "name": "c3.4xlarge",
        "pretty_name": "C3 High-CPU Quadruple Extra Large",
        "memory": 30,
        "vcpus": 16
    },
    {
        "name": "r4.4xlarge",
        "pretty_name": "R4 High-Memory Quadruple Extra Large",
        "memory": 122,
        "vcpus": 16
    },
    {
        "name": "r5a.xlarge",
        "pretty_name": "R5A Extra Large",
        "memory": 32,
        "vcpus": 4
    },
    {
        "name": "h1.2xlarge",
        "pretty_name": "H1 Double Extra Large",
        "memory": 32,
        "vcpus": 8
    },
    {
        "name": "m5ad.24xlarge",
        "pretty_name": "M5AD 24xlarge",
        "memory": 384,
        "vcpus": 96
    },
    {
        "name": "i3en.xlarge",
        "pretty_name": "I3EN Extra Large",
        "memory": 32,
        "vcpus": 4
    },
    {
        "name": "r5ad.24xlarge",
        "pretty_name": "R5AD 24xlarge",
        "memory": 768,
        "vcpus": 96
    },
    {
        "name": "g2.2xlarge",
        "pretty_name": "G2 Double Extra Large",
        "memory": 15,
        "vcpus": 8
    },
    {
        "name": "c4.2xlarge",
        "pretty_name": "C4 High-CPU Double Extra Large",
        "memory": 15,
        "vcpus": 8
    },
    {
        "name": "m5.4xlarge",
        "pretty_name": "M5 General Purpose Quadruple Extra Large",
        "memory": 64,
        "vcpus": 16
    },
    {
        "name": "z1d.xlarge",
        "pretty_name": "Z1D Extra Large",
        "memory": 32,
        "vcpus": 4
    },
    {
        "name": "t3.2xlarge",
        "pretty_name": "T3 Double Extra Large",
        "memory": 32,
        "vcpus": 8
    },
    {
        "name": "m4.2xlarge",
        "pretty_name": "M4 General Purpose Double Extra Large",
        "memory": 32,
        "vcpus": 8
    },
    {
        "name": "r5ad.large",
        "pretty_name": "R5AD Large",
        "memory": 16,
        "vcpus": 2
    },
    {
        "name": "r5.4xlarge",
        "pretty_name": "R5 Quadruple Extra Large",
        "memory": 128,
        "vcpus": 16
    },
    {
        "name": "r5d.2xlarge",
        "pretty_name": "R5D Double Extra Large",
        "memory": 64,
        "vcpus": 8
    },
    {
        "name": "r5.large",
        "pretty_name": "R5 Large",
        "memory": 16,
        "vcpus": 2
    },
    {
        "name": "i3.large",
        "pretty_name": "I3 High I/O Large",
        "memory": 15.25,
        "vcpus": 2
    },
    {
        "name": "z1d.large",
        "pretty_name": "Z1D Large",
        "memory": 16,
        "vcpus": 2
    },
    {
        "name": "m5ad.12xlarge",
        "pretty_name": "M5AD 12xlarge",
        "memory": 192,
        "vcpus": 48
    },
    {
        "name": "i3en.large",
        "pretty_name": "I3EN Large",
        "memory": 16,
        "vcpus": 2
    },
    {
        "name": "c5d.9xlarge",
        "pretty_name": "C5 High-CPU 9xlarge",
        "memory": 72,
        "vcpus": 36
    },
    {
        "name": "r5.xlarge",
        "pretty_name": "R5 Extra Large",
        "memory": 32,
        "vcpus": 4
    },
    {
        "name": "i3.2xlarge",
        "pretty_name": "I3 High I/O Double Extra Large",
        "memory": 61,
        "vcpus": 8
    },
    {
        "name": "r5d.large",
        "pretty_name": "R5D Large",
        "memory": 16,
        "vcpus": 2
    },
    {
        "name": "g3.4xlarge",
        "pretty_name": "G3 Quadruple Extra Large",
        "memory": 122,
        "vcpus": 16
    },
    {
        "name": "i3.4xlarge",
        "pretty_name": "I3 High I/O Quadruple Extra Large",
        "memory": 122,
        "vcpus": 16
    },
    {
        "name": "c5n.4xlarge",
        "pretty_name": "C5N Quadruple Extra Large",
        "memory": 42,
        "vcpus": 16
    },
    {
        "name": "r3.4xlarge",
        "pretty_name": "R3 High-Memory Quadruple Extra Large",
        "memory": 122,
        "vcpus": 16
    },
    {
        "name": "m1.large",
        "pretty_name": "M1 General Purpose Large",
        "memory": 7.5,
        "vcpus": 2
    },
    {
        "name": "m2.4xlarge",
        "pretty_name": "M2 High Memory Quadruple Extra Large",
        "memory": 68.4,
        "vcpus": 8
    },
    {
        "name": "m2.xlarge",
        "pretty_name": "M2 High Memory Extra Large",
        "memory": 17.1,
        "vcpus": 2
    },
    {
        "name": "t3a.2xlarge",
        "pretty_name": "T3A Double Extra Large",
        "memory": 32,
        "vcpus": 8
    },
    {
        "name": "c1.xlarge",
        "pretty_name": "C1 High-CPU Extra Large",
        "memory": 7,
        "vcpus": 8
    },
    {
        "name": "m5.8xlarge",
        "pretty_name": "M5 General Purpose Eight Extra Large",
        "memory": 128,
        "vcpus": 32
    },
    {
        "name": "m2.2xlarge",
        "pretty_name": "M2 High Memory Double Extra Large",
        "memory": 34.2,
        "vcpus": 4
    },
    {
        "name": "r5ad.4xlarge",
        "pretty_name": "R5AD Quadruple Extra Large",
        "memory": 128,
        "vcpus": 16
    },
    {
        "name": "r5ad.12xlarge",
        "pretty_name": "R5AD 12xlarge",
        "memory": 384,
        "vcpus": 48
    },
    {
        "name": "m5a.large",
        "pretty_name": "M5A Large",
        "memory": 8,
        "vcpus": 2
    },
    {
        "name": "c4.4xlarge",
        "pretty_name": "C4 High-CPU Quadruple Extra Large",
        "memory": 30,
        "vcpus": 16
    },
    {
        "name": "t3.large",
        "pretty_name": "T3 Large",
        "memory": 8,
        "vcpus": 2
    },
    {
        "name": "g3s.xlarge",
        "pretty_name": "G3S Extra Large",
        "memory": 30.5,
        "vcpus": 4
    },
    {
        "name": "a1.4xlarge",
        "pretty_name": "A1 Quadruple Extra Large",
        "memory": 32,
        "vcpus": 16
    },
    {
        "name": "f1.2xlarge",
        "pretty_name": "F1 Double Extra Large",
        "memory": 122,
        "vcpus": 8
    },
    {
        "name": "t2.medium",
        "pretty_name": "T2 Medium",
        "memory": 4,
        "vcpus": 2
    },
    {
        "name": "a1.large",
        "pretty_name": "A1 Large",
        "memory": 4,
        "vcpus": 2
    },
    {
        "name": "t3a.xlarge",
        "pretty_name": "T3A Extra Large",
        "memory": 16,
        "vcpus": 4
    },
    {
        "name": "u-6tb1.metal",
        "pretty_name": "U-6TB1 Metal",
        "memory": 6144,
        "vcpus": 448
    },
    {
        "name": "t3a.large",
        "pretty_name": "T3A Large",
        "memory": 8,
        "vcpus": 2
    },
    {
        "name": "u-9tb1.metal",
        "pretty_name": "U-9TB1 Metal",
        "memory": 9216,
        "vcpus": 448
    },
    {
        "name": "u-12tb1.metal",
        "pretty_name": "U-12TB1 Metal",
        "memory": 12288,
        "vcpus": 448
    }
]
```

**Documentation**:

**Does this PR introduce a user-facing change?**:

```release-note
Added endpoint to list AWS sizes.
```

### CODE REVIEW

File: allowed_licensed.yaml [hunk 1]
```diff
@@ -19,3 +19,4 @@ exceptions:
   - bitbucket.org/ww/goautoneg
   - github.com/ajeddeloh/go-json
   - github.com/opencontainers/go-digest # Cannot detect the license for some reason. Its Apache 2.0
+  - github.com/cristim/ec2-instances-info # Public domain and MIT. See: https://github.com/cristim/ec2-instances-info/blob/master/LICENSE and https://github.com/powdahound/ec2instances.info/blob/master/LICENSE.
```
Comment (maciaszczykm) [inline allowed_licensed.yaml]: @kdomanski Can we use it? The repo code is "public domain" and the generated data is under MIT from what I can see. Is that okay?

File: api/pkg/handler/v1/provider/aws.go [hunk 2]
```diff
@@ -134,6 +135,39 @@ func DecodeAWSVPCReq(c context.Context, r *http.Request) (interface{}, error) {
...
+			return nil, err
+		}
+
+		sizes := apiv1.AWSSizeList{}
+		for _, i := range *data {
+			// TODO: Make the check below more generic, working for all the providers. It is needed as the pods
+			//  with memory under 2 GB will be full with required pods like kube-proxy, CNI etc.
+			if i.Memory > 2 {
+				// The code below filters out too expensive instance types (>2$ per hour).
+				// TODO: Use correct datacenter to get the pricing and enable changing 2$ price limit.
```
Comment (maciaszczykm) [inline api/pkg/handler/v1/provider/aws.go]: We would need to pass region here, not the name of the seed cluster.

File: allowed_licensed.yaml (hunk 1 above)
Comment (kdomanski) [reply allowed_licensed.yaml]: That's okay.

Comment (maciaszczykm) [reply allowed_licensed.yaml]: Okay, thank you.

File: api/pkg/handler/v1/provider/aws.go [hunk 3]
```diff
@@ -64,6 +66,18 @@ type AWSVPCReq struct {
 	DC string `json:"dc"`
 }
 
+// AzureSizeReq represent a request for Azure VM sizes
```
Comment (zreigz) [inline api/pkg/handler/v1/provider/aws.go]: change `Azure` to `AWS`

File: api/pkg/handler/v1/provider/aws.go [hunk 4]
```diff
@@ -64,6 +66,18 @@ type AWSVPCReq struct {
...
+// swagger:parameters listAWSSizesNoCredentials
+type AWSSizeReq struct {
+	Region string
+}
+
+func DecodeAWSSizesReq(c context.Context, r *http.Request) (interface{}, error) {
```
Comment (zreigz) [inline api/pkg/handler/v1/provider/aws.go:76]: add documentation comments

**Review by zreigz — CHANGES_REQUESTED** [review]

**Pull request merged**